provisioning/hare_setup.py

```diff
@@ -0,0 +1,383 @@
+                          error)
+            exit(-1)
+
+class Init(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            if is_cluster_running() == True:
+                logging.debug('Cluster is already running')
+                if shutdown_cluster() != 0:
+                    logging.error('Failed to shutdown the cluster')
+                    exit(-1)
+                
+            if bootstrap_cluster() != 0:
+                logging.error('Failed to bootstrap the cluster')
+                exit(-1)
+            if shutdown_cluster() != 0:
+                logging.error('Failed to shutdown the cluster')
+                exit(-1)
+        except Exception as error:
+            logging.error('Error while initializing the cluster (%s)',
+                          error)
+            exit(-1)
+
+class Test(argparse.Action):
+    """
+    Read CDF file and compare fields with output of 'hctl status'
+    Will start the cluster(and shutdown before exiting) if not already started
+    """
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            cluster_started = False
+            cluster_status_failure = False
+            if is_cluster_running() == False:
+                if bootstrap_cluster() != 0:
+                    logging.error('Failed to bootstrap the cluster')
+                    exit(-1)
+                cluster_started = True
+
+            if check_cluster_status() != 0:
+                logging.error('Cluster status reports failure')
+                cluster_status_failure = True
+
+            if cluster_started == True:
+                if shutdown_cluster() != 0:
+                    logging.error('Failed to shutdown the cluster')
+                    exit(-1)
+
+            if cluster_status_failure == True:
+                exit(-1)
+
+        except Exception as error:
+            logging.error('Error while checking cluster status (%s)',
+                          error)
+            exit(-1)
+
+class SupportBundle(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            # Default target directory is /tmp/hare
+            if os.system('hctl reportbug') != 0:
+                logging.error('Failed to generate support bundle')
+                exit(-1)
+        except Exception as error:
+            logging.error('Error while generating support bundle (%s)',
+                          error)
+            exit(-1)
+
+class Cleanup(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            logging.info('No-op for Cleanup/Reset as of now for Hare ')
+        except Exception as error:
+            logging.error('Failed to clear Hare logs (%s)',
+                          error)
+            exit(-1)
+
+def checkRpm(rpm_name):
+    return os.system('rpm -qa | grep -q ' + rpm_name)
```

Comment:
@SwapnilGaonkar7 If you're curious, there is a way to invoke piped commands using `subprocess`. Probably that will make Codacy happier.

Here is the example how it can be done in our Hare commit history: https://github.com/Seagate/cortx-hare/blob/4e360eec9b5501d26af7ded61ad139691283c8e6/ha-notifier/bq/listener.py#L39

There, I invoke `m0hagen | m0ham` through this api.

Reply:
Done